nelox/interpreter.py

```diff
@@ -0,0 +1,223 @@
+
+
+def _sub(*args):
+    if not args:
+        raise RuntimeError("'-' requires at least one argument")
+
+    result = args[0]
+
+    for arg in args[1:]:
+
+        if isinstance(result, (int, float)) and isinstance(arg, (int, float)):
+            result -= arg
+
+        elif isinstance(result, list) and isinstance(arg, (int, float)):
+            result = [x - arg for x in result]
+
+        elif isinstance(result, (int, float)) and isinstance(arg, list):
+            result = [result - x for x in arg]
+
+        else:
+            raise RuntimeError(f"Unsupported operands for '-': {type(result)} and {type(arg)}")
+    return result
+
+
+def _mult(*args):
+    if not args:
+        raise RuntimeError("'*' requires at least one argument")
+
+    result = args[0]
+    for arg in args[1:]:
+
+        if isinstance(result, list) and isinstance(arg, (int, float)):
+            result = [x * arg for x in result]
+        elif isinstance(result, (int, float)) and isinstance(arg, list):
+            result = [result * x for x in arg]
+
+        elif isinstance(result, (int, float)) and isinstance(arg, (int, float)):
+            result = result * arg
+
+        else:
+            raise RuntimeError(f"Unsupported operands for '*': {type(result)} and {type(arg)}")
+
+    return result
+
+
+def _div(*args):
+    if not args:
+        raise RuntimeError("'/' requires at least one argument")
+
+    result = args[0]
+
+    for arg in args[1:]:
+
+        if isinstance(result, (int, float)) and isinstance(arg, (int, float)):
+            result /= arg
+
+        elif isinstance(result, list) and isinstance(arg, (int, float)):
+            result = [x / arg for x in result]
+
+        elif isinstance(result, (int, float)) and isinstance(arg, list):
+            result = [result / x for x in arg]
+
+        else:
+            raise RuntimeError(f"Unsupported operands for '/': {type(result)} and {type(arg)}")
+    return result
+
+
+def _builtin_get(lst, index):
+    if not isinstance(lst, list):
+        raise RuntimeError(f"'get': expected list, got {type(lst)}")
+    if not isinstance(index, int):
+        raise RuntimeError(f"'get': expected integer index, got {type(index)}")
+    try:
+        return lst[index]
+    except IndexError:
+        raise RuntimeError("'get': index out of bounds")
+
+
+def _builtin_length(lst):
+    if not isinstance(lst, list):
+        raise RuntimeError(f"'length': expected list, got {type(lst)}")
+    return len(lst)
+
+
+def _define_builtins(env):
+    env.set("list", lambda *args: list(args))
+    env.set("get", _builtin_get)
+    env.set("length", _builtin_length)
+    env.set("true", True)
+    env.set("false", False)
+    env.set("+", _add)
+    env.set("-", _sub)
+    env.set("*", _mult)
+    env.set("/", _div)
+    env.set(">", lambda a, b: a > b)
+    env.set("<", lambda a, b: a < b)
+    env.set(">=", lambda a, b: a >= b)
+    env.set("<=", lambda a, b: a <= b)
+    env.set("=", lambda a, b: a == b)
```

Comment:
Similarly: perhaps these should operate on lists?  It would be good to have a general philosophy of how your language works.